test/test_datapipe.py

```diff
@@ -77,10 +140,14 @@ def setUp(self):
         self.temp_sub_dir = ret[1][0]
         self.temp_sub_files = ret[1][1:]
 
+        set_up_local_server()
+        create_files_to_web_temp()
```

Comment:
Can you create a separate TestCase class, or just move these methods into your test methods? IIRC, every single test method will call setup and tearDown, which should not be expected and too redundant in this case.

Reply:
Separate TestCase class with proper seUp and tearDown is the best here

Reply:
thank you for all comments! will change as suggested

Reply:
fyi it is now moved to a new separate testcase TestIterableDataPipeHttp.